pyscf/neo/hf.py

```diff
@@ -378,17 +378,119 @@ def _build_eri(mol1, mol2, cart):
                                              size1, size1 + size2),
                                  aosym='s4')
 
+def cholesky_eneri(mol_n, auxmol_e, int3c, int2c, max_memory,
+                   aosym='s2ij', decompose_j2c='cd'):
+    from pyscf.df.incore import _eig_decompose
+    from pyscf.df.outcore import _guess_shell_ranges
+    j2c_e = auxmol_e.intor(int2c, hermi=1)
+    if decompose_j2c == 'eig':
+        low = _eig_decompose(auxmol_e, j2c_e)
+    else:
+        try:
+            low = scipy.linalg.cholesky(j2c_e, lower=True)
```

Comment:
This step may become the bottleneck if you have 1000+ auxbasis functions. Need to test the timings! If this becomes the bottleneck, should share the cholesky decomposition from e-e df.